src/core/sys/netbsd/dlfcn.d

```diff
@@ -0,0 +1,114 @@
+/**
+ * D header file for NetBSD.
+ *
+ * Copyright: Copyright Martin Nowak 2012.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Martin Nowak
+ *
+ * http://cvsweb.netbsd.org/bsdweb.cgi/~checkout~/src/include/dlfcn.h
+ */
+module core.sys.netbsd.dlfcn;
+
+public import core.sys.posix.dlfcn;
+
+version (NetBSD):
+extern (C):
+nothrow:
+
+enum __BSD_VISIBLE = true;
+
+/*
+ * Modes and flags for dlopen().
+ */
+static assert(RTLD_LAZY   == 1);
+static assert(RTLD_NOW    == 2);
+static assert(RTLD_GLOBAL == 0x100);
+static assert(RTLD_LOCAL  == 0x200);
+//enum RTLD_TRACE           =  0x200;
+enum RTLD_NODELETE        =  0x01000;
+enum RTLD_NOLOAD          =  0x02000;
+
+/*
+ * Request arguments for dlinfo().
+ */
+enum RTLD_DI_LINKMAP     = 3;    /* Obtain link map. */
+enum RTLD_DI_SERINFO     = 5;    /* Obtain search path info. */
+enum RTLD_DI_SERINFOSIZE = 6;    /*  ... query for required space. */
+enum RTLD_DI_ORIGIN      = 7;    /* Obtain object origin */
+enum RTLD_DI_MAX         = RTLD_DI_ORIGIN;
+
+/*
+ * Special handle arguments for dlsym()/dlinfo().
+ */
+enum RTLD_NEXT    = cast(void *)-1;    /* Search subsequent objects. */
+enum RTLD_DEFAULT = cast(void *)-2;    /* Use default search algorithm. */
+enum RTLD_SELF    = cast(void *)-3;    /* Search the caller itself. */
+
+static if (__BSD_VISIBLE)
+{
+    /*
+     * Structure filled in by dladdr().
+     */
+    struct Dl_info {
+        const(char)     *dli_fname;     /* Pathname of shared object. */
+        void            *dli_fbase;     /* Base address of shared object. */
+        const(char)     *dli_sname;     /* Name of nearest symbol. */
+        void            *dli_saddr;     /* Address of nearest symbol. */
+    };
+
+    /*-
+     * The actual type declared by this typedef is immaterial, provided that
+     * it is a function pointer.  Its purpose is to provide a return type for
+     * dlfunc() which can be cast to a function pointer type without depending
+     * on behavior undefined by the C standard, which might trigger a compiler
+     * diagnostic.  We intentionally declare a unique type signature to force
+     * a diagnostic should the application not cast the return value of dlfunc()
+     * appropriately.
+     */
+    struct __dlfunc_arg {
+        int     __dlfunc_dummy;
+    };
+
+    alias void function(__dlfunc_arg) dlfunc_t;
```

Comment:
Can you change this to the `alias =` syntax, as now done in the same FreeBSD module?  Same with the `__externC` alias below.